src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php

```diff
@@ -826,18 +823,16 @@ public function enterHash(Cursor $cursor, int $type, $class, bool $hasChild)
         }
         parent::enterHash($cursor, $type, $class, false);
 
-        if ($cursor->skipChildren) {
+        if ($cursor->skipChildren || $cursor->depth >= $this->displayOptions['maxDepth']) {
             $cursor->skipChildren = false;
             $eol = ' class=sf-dump-compact>';
-        } elseif ($this->expandNextHash) {
+        } else {
```

Comment:
wait, now this should be kept since you removed the `||` that you added in the 1st iteration :)

Reply:
Since the decision on rendering compact/extended has been removed from a front end, we know in what state each element is in. We no longer need to render elements without the state.